src/PolygonClipper/PolygonClipper.cs

```diff
@@ -622,21 +635,33 @@ private static int PossibleIntersection(
         // Handle the rightCoincide case
         if (rightCoincide)
         {
-            DivideSegment(events[0], events[1].Point, eventQueue, comparer);
+            ArgumentNullException.ThrowIfNull(first);
+            ArgumentNullException.ThrowIfNull(second);
+
+            DivideSegment(first, second.Point, eventQueue, comparer);
             return 3;
         }
 
+        ArgumentNullException.ThrowIfNull(fourth);
+
         // Handle general overlapping case
-        if (events[0] != events[3].OtherEvent)
+        if (first != fourth.OtherEvent)
         {
-            DivideSegment(events[0], events[1].Point, eventQueue, comparer);
-            DivideSegment(events[1], events[2].Point, eventQueue, comparer);
+            ArgumentNullException.ThrowIfNull(first);
```

Comment:
All these guard checks would kill performance as we're calling this method from a while loop.

Reply:
Good to know. That was the NRE brain doing its work 😊